### ISSUE TITLE
Tensorflow feed_dict issue

### ISSUE THREAD

**Issue by XiminLin** [tensorflow/tensorflow#11542]
I am new to tensorflow. I understand that we need to create the tensorflow graph and then call sess.run() to get the values I want.

However, I am confused by how the feed_dict works. 
For example:
Will 1 be the same as 2?
```
1. accuracy, cost = sess.run([accuracy, cost], feed_dict = {X:X_batch, Y:Y_batch})

2. accuracy  = sess.run(accuracy, feed_dict = {X:X_batch, Y: Y_batch})
   cost = sess.run(cost, feed_dict = {X:X_batch, Y:Y_batch})
```
I don't know that if tensorflow receives the same feed_dict in cost and in tensorflow graph computing accuracy already computes cost, do it go over the neural net again to evaluate the value, or it will return the value computed without going through the net again?

Also, from [Hvass-Labs/TensorFlow-Tutorials/TensorFlow Tutorial #02 Convolutional Neural Network](https://github.com/Hvass-Labs/TensorFlow-Tutorials/blob/master/02_Convolutional_Neural_Network.ipynb), in function plot_conv_weights(weights, input_channel=0)
```
weights = sess.run(conv_weigh)
```
Since training weights require we fill the placeholder X and Y with values, but here I saw no feed_dict.

So how exactly feed_dict works?

**Comment by asimshankar** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

That said, the short answer is that the two should be the same, assuming that in the course of computing `accuracy` and/or `cost`, the model weights are not modified. In general, each call to `Session.run` will execute all the computations required to compute the requested tensors to fetch. See https://www.tensorflow.org/api_docs/python/tf/Session#run
